File: ruby/rna-transcription/rna_transcription.rb
module Complement
  DNA_TO_RNA = {
    "G" => "C",
    "C" => "G",
    "T" => "A",
    "A" => "U",
  }

  def self.of_dna(dna)
    rna = String.new

    dna.each_char do |nucleotide|
      rna << DNA_TO_RNA[nucleotide]
    end

    rna
  end
end

File: cpp/rna-transcription/rna_transcription.cpp
#include "rna_transcription.h"

using namespace std;

namespace rna_transcription
{
    string to_rna(string dna)
    {
        for (char &c : dna)
        {
            c = to_rna(c);
        }

        return dna;
    }

    char to_rna(char dna)
    {
        switch (dna)
        {
        case 'G':
            return 'C';

        case 'C':
            return 'G';

        case 'T':
            return 'A';

        case 'A':
            return 'U';

        default:
            return dna;
        }
    }
} // namespace rna_transcription
